### PR TITLE
PLT-1557 revoking sessions from same device

### PR DESCRIPTION
**Pull request by coreyhulen**

### CODE REVIEW

File: api/user.go [hunk 1]
```diff
@@ -497,6 +497,26 @@ func Login(c *Context, w http.ResponseWriter, r *http.Request, user *model.User,
...
+		// A special case where we logout of all other sessions with the same Id
+		if result := <-Srv.Store.Session().GetSessions(user.Id); result.Err != nil {
+			c.Err = result.Err
+			c.Err.StatusCode = http.StatusForbidden
+			return
+		} else {
+			sessions := result.Data.([]*model.Session)
+			for _, session := range sessions {
+				if session.DeviceId == deviceId {
+					l4g.Debug("Revoking sessionId=" + session.Id + " for userId=" + user.Id + " re-login with same device Id")
```
Comment (jwilander) [inline api/user.go]: 0/5 but should this be l4g.Info or l4g.Warn instead of l4g.Debug?

**Comment by jwilander** [comment]
+1, other than minor comment

File: api/user.go (hunk 1 above)
Comment (coreyhulen) [inline api/user.go]: RevokeSessionById calls Audit(), this is just extra information for debugging.

Comment (jwilander) [inline api/user.go]: Sounds good

File: api/user.go [hunk 2]
```diff
@@ -664,13 +684,15 @@ func loginLdap(c *Context, w http.ResponseWriter, r *http.Request) {
 func revokeSession(c *Context, w http.ResponseWriter, r *http.Request) {
```
Comment (crspeller) [inline api/user.go]: Better name for this would be revokeCurrentSession.

Comment (coreyhulen) [inline api/user.go]: @crspeller This is the existing function and you can revoke any session, not just the current one.

**Comment by crspeller** [comment]
+1

**Pull request merged**